cmd/collector/app/span_handler_builder.go

```diff
@@ -36,7 +36,7 @@ type SpanHandlers struct {
 }
 
 // BuildSpanProcessor builds the span processor to be used with the handlers
-func (b *SpanHandlerBuilder) BuildSpanProcessor(additional ...ProcessSpan) processor.SpanProcessor {
+func (b *SpanHandlerBuilder) BuildSpanProcessor(additional ...ProcessSpan) (processor.SpanProcessor, error) {
```

Comment:
@yurishkuro why are we adding error to the signature here if this function isn't returning an error? is there going to be a follow-up PR?

Reply:
yes - as explained in the PR description, introducing OTEL components will introduce errors